src/main/java/net/countercraft/movecraft/combat/movecraftcombat/directors/CannonDirectorManager.java

```diff
@@ -122,11 +123,29 @@ private void processDirectors() {
                 if (!(tnt.getVelocity().lengthSquared() > 0.35) || tracking.containsKey(tnt)) {
                     continue;
                 }
-                Craft c = CraftManager.getInstance().fastNearestCraftToLoc(tnt.getLocation());
+
+                Craft c;
+
+                if(Config.EnableTNTTracking) {
+                    if(!tnt.hasMetadata("MCC-Sender")) {
+                        continue;
+                    }
+                    Player sender = Bukkit.getPlayer(UUID.fromString(tnt.getMetadata("MCC-Sender").get(0).asString()));
```

Comment:
This method has a chance to cause an exception.  See 2f8ec3a2139f12348a3f7d5fce06dff963d302fb for the fix.